raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -893,7 +923,12 @@ def handle_init(
     if not payer_channel:
         return TransitionResult(None, [])
 
-    mediator_state = MediatorTransferState(from_transfer.lock.secrethash)
+    mediator_state = MediatorTransferState(
+        payer_channel_identifier=payer_channel.identifier,
+        payer_transfer=from_transfer,
```

Comment:
This is insufficient. It's perfectly possible for the following topology:

```
     - C -
A - B     D
     - E -
```

For A to send a transfer to D through B while E is offline, forcing B to C route to C which can refund.

Meaning instead of saving the *first received* we need the *latest* which **is** waiting.